docs/learning/guidelines/release-tags-guidelines.md

```diff
@@ -1,66 +1,205 @@
-API review files can then be placed under source code control and compared to future SDK versions to ensure that the API is evolving in compatible or expected ways.
+iTwin.js uses [release tags](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags) to classify individual APIs according to their intended level of support. Consumers of iTwin.js should consult [API support policies](../api-support-policies.md) for an overview. This article provides iTwin.js contributors with guidance on applying release tags.
 
-## Supported Release Tags
+## Supported tags
 
-The four primary release tags and the "deprecated" tag are described in the [API support policies](../api-support-policies.md). For [Extensions](../frontend/Extensions.md), two additional tags are supported:
+Each API in iTwin.js belongs to one of four API categories as defined by its associated release tag:
 
-- The "extensions" tag indicates that an API is to be included in the `@itwin/core-extension` API. Currently this tag is only useful for APIs in `@itwin/core-frontend` and `@itwin/core-common`; all other packages' APIs can be used directly by extensions. It may become relevant for more packages in the future. A lint rule enforces that the "extensions" tag may only be applied to APIs tagged as "public".
+- `@public` indicates a stable, production-ready API subject to our documented [stability and deprecation policies](../api-support-policies.md).
+- `@beta` indicates an API undergoing stabilization. Users are encouraged to experiment with it in non-production environments.
+- `@alpha` indicates a highly experimental API that users should not yet experiment with.
+- `@internal` indicates an API that should never be used outside of the itwinjs-core package.
 
-## API Items
+By default, any API is also usable by extensions, with the exception of APIs exported by `@itwin/core-frontend` and `@itwin/core-common`; these must be explicitly tagged as `@extensions` for inclusion in the `@itwin/core-extension` package's API. The `@extensions` tag may only be applied to `@public` APIs, as verified by a lint rule.
 
-Every API exported by a package must have a release tag.
-Members of classes, interface, namespace, and enums inherit the release tag applied to their containing type unless explicitly overridden.
-The release tag applied to such a member can only be more restrictive than that applied to its container. For example, a "beta" class can contain "alpha" or "internal" properties, but not "public" ones.
+The `@deprecated` tag can be applied to any API to notify users that it may be removed or changed in a breaking way in the future. For `@public` APIs, deprecations must follow our [deprecation policy](../api-support-policies.md#api-deprecation-policy); marking `@beta`, `@alpha`, or `@internal` APIs as deprecated is optional, but recommended for giving users a heads-up.
```

Comment:
Should this mention the requirement to specify an alternative API to use instead of the deprecated one?

P.S. I find the idea to deprecate with intention to change in a breaking way, rather than remove the API, quite interesting. Technically, that's just removing + adding a different API with the same name, but from consumer's perspective we're suggesting them to switch from using `X` to `Y` and then start using `X` again.

Reply:
That - along with the requirement to specify the version in which it was deprecated - is documented above the code example.